mlflow/utils/environment.py

```diff
@@ -80,5 +84,61 @@ def _get_additional_pip_dep(conda_env):
 
 
 def _log_pip_requirements(conda_env, path, requirements_file="requirements.txt"):
-    pip_deps = _get_additional_pip_dep(conda_env)
+    pip_deps = _get_pip_dep(conda_env)
     _mlflow_additional_pip_env(pip_deps, path=os.path.join(path, requirements_file))
+
+
+def _parse_pip_requirements(pip_requirements):
+    def _is_string(x):
+        return isinstance(x, str)
+
+    def _is_iterable(x):
+        try:
+            iter(x)
+            return True
+        except Exception:
+            return False
+
+    if _is_string(pip_requirements):
+        return [
+            req.requirement for req in parse_requirements(pip_requirements, session=PipSession())
+        ]
+    elif _is_iterable(pip_requirements) and all(map(_is_string, pip_requirements)):
+        try:
+            # Create a temporary requirements file in the current working directory
+            requirements = tempfile.NamedTemporaryFile(
+                mode="w",
+                prefix="mlflow.",
+                suffix=".tmp.requirements.txt",
+                dir=os.getcwd(),
+                delete=False,
+            )
```

Comment:
Nit: We can use [mlflow.utils.file_utils.TempDir](https://github.com/mlflow/mlflow/blob/bc86f1dfdb6444746470b7b12b6582b1390eb1aa/mlflow/utils/file_utils.py#L183) to avoid the try/catch

Reply:
Thanks, I'll use `mlflow.utils.file_utils.TempDir` with `chdr`.

Reply:
Found an issue related to the comment:

https://bugs.python.org/issue14243

Reply:
Hmm `mlflow.utils.file_utils.TempDir` doesn't seem to work because we need to create a temporary requirements file in the current working directory